Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php

```diff
@@ -238,32 +264,98 @@ protected function save()
-                $this->getObject()->deleteAutomaticMembershipSource((string) $type, (int) $id);
-            }
+
+        if (!array_key_exists($field, $post)) {
+            ilUtil::sendInfo($this->lng->txt('prg_delete_nothing_selected'), true);
+            $this->ctrl->redirect($this, self::CMD_VIEW);
+        }
+
+        $str = '';
+        foreach ($post[$field] as $data) {
+            $str .= $data . ' ';
         }
+
+        $str = base64_encode($str);
+        $message = $this->lng->txt('prg_delete_confirmation');
+        $this->ctrl->setParameterByClass(self::class, 'type_ids', $str);
+        $delete = $this->ctrl->getFormActionByClass(self::class, self::CMD_DELETE);
+        $cancel = $this->ctrl->getFormActionByClass(self::class, self::CMD_VIEW);
+        $this->ctrl->clearParameterByClass(self::class, 'type_ids');
+
+        $buttons = [
+            $this->button_factory->standard($this->lng->txt('prg_confirm_delete'), $delete),
+            $this->button_factory->standard($this->lng->txt('prg_cancel'), $cancel)
+        ];
+
+        $message_box = $this->message_box_factory->confirmation($message)->withButtons($buttons);
+
+        $this->tpl->setContent($this->ui_renderer->render($message_box));
+    }
+
+    protected function delete()
+    {
+        $get = $_GET;
+        if (!array_key_exists('type_ids', $get)) {
+            ilUtil::sendFailure($this->lng->txt('prg_delete_failure'), true);
+            $this->ctrl->redirect($this, self::CMD_VIEW);
+        }
+
+        $type_ids = base64_decode($get['type_ids']);
+
+        $type_ids = explode(' ', trim($type_ids));
+
+        foreach ($type_ids as $src_id) {
+            [$type, $id] = explode('-', $src_id);
+            $this->getObject()->deleteAutomaticMembershipSource((string) $type, (int) $id);
+        }
+
+        ilUtil::sendSuccess($this->lng->txt('prg_delete_success'), true);
+        $this->ctrl->redirect($this, self::CMD_VIEW);
+    }
+
+    protected function deleteSingleConfirmation()
+    {
+        $get = $_GET;
+        $field = self::CHECKBOX_SOURCE_IDS;
+        if (!array_key_exists($field, $get)) {
+            ilUtil::sendFailure($this->lng->txt('prg_delete_single_failure'), true);
+            $this->ctrl->redirect($this, self::CMD_VIEW);
+        }
+
+        $message = $this->lng->txt('prg_delete_single_confirmation');
+        $this->ctrl->setParameterByClass(self::class, $field, $get[$field]);
+        $delete = $this->ctrl->getFormActionByClass(self::class, self::CMD_DELETE_SINGLE);
+        $cancel = $this->ctrl->getFormActionByClass(self::class, self::CMD_VIEW);
+        $this->ctrl->clearParameterByClass(self::class, $field);
+
+        $buttons = [
+            $this->button_factory->standard($this->lng->txt('prg_confirm_delete'), $delete),
+            $this->button_factory->standard($this->lng->txt('prg_cancel'), $cancel)
+        ];
+
+        $message_box = $this->message_box_factory->confirmation($message)->withButtons($buttons);
```

Comment:
This code seems to be mostly duplicated with `deleteConfirmation`. Can we unify it?

Reply:
done